test/extensions/transport_sockets/tls/test_private_key_method_provider.cc

```diff
@@ -257,6 +257,8 @@ bool TestPrivateKeyMethodProvider::checkFips() {
   return true;
 }
 
+bool TestPrivateKeyMethodProvider::initialized() { return true; }
```

Comment:
It would be great to have a test to utilize this `TestPrivateKeyMethodProvider` to ensure when the fallback case, this TestPrivateKeyMethodProvider won't be used. Probably you can copy this test https://github.com/envoyproxy/envoy/blob/5fd5f1281ddad25f893a74241a0a4f316cfe2191/test/extensions/transport_sockets/tls/ssl_socket_test.cc#L6273 and with some modification, then you should be able to test the case what I said.

Reply:
done, thanks!